docs/guides/ajax.rst

```diff
@@ -1,31 +1,234 @@
 Ajax
 ####
 
-Actions
-=======
+The ``elgg/Ajax`` AMD module (introduced in Elgg 2.1) provides a set of methods for communicating with the server in a concise and uniform way, which allows plugins to collaborate on the request data, the server response, and the returned client-side data.
 
-From JavaScript we can execute actions via XHR POST operations. Here's an example action and script for some basic math:
+Client and server code written for the legacy API should not need modification.
 
-.. code:: php
+.. contents:: Contents
+   :local:
+   :depth: 2
 
-    // in myplugin/actions/do_math.php
+Overview
+========
 
-    if (!elgg_is_xhr()) {
-        register_error('Sorry, Ajax only!');
-        forward();
-    }
+All the ajax methods perform the following actions (not the hook types depend on the method):
 
-    $arg1 = (int)get_input('arg1');
-    $arg2 = (int)get_input('arg2');
+#. Client-side, the ``data`` option is filtered by the hook ``ajax_request_data``.
```

Comment:
Perhaps `ajax_request_data, {action name}`? (Or `ajax_request_data, <action name>}`? Which type of brackets do we usually use?)